nautobot/docs/models/extras/dynamicgroup.md

```diff
@@ -1,7 +1,166 @@
+
+Consider an example where there is a graph from the parent group to three direct child groups, the third of which has its own nested child group:
+
+```no-highlight
+parent
+- first-child
+- second-child
+- third-child
+  - nested-child
+```
+
+The filter generation would walk the graph topologically, starting from the base filter of `parent`, the filter of `first-child` would be applied to that of `parent`, then `second-child`, in order, all the way down. In the case of `third-child`, all of its children (only `nested-child` in this case) would be processed in order in the same way and the resultant filter from all of the child groups for `third-child` would be used to process the filter that resulted from the filter of `second-child`.
+
+If this is confusing, don't worry. We'll cover it more in hands-on examples after this section.
+
+### Weights
+
+Weights are used to enforce the topological sorting of how filters are processed when traversing from a parent to each descending child group. Because this ordering is significant, care must be taken when constructing nested Dynamic Groups to result in filter parameters that have the desired outcome.
+
+In practice, weights are automatically assigned in increments of `10`. In the web UI, child groups may be dragged and dropped to explicitly sort them.
+
+Using the example group hierarchy above, the weights would be as follows:
+
+```no-highlight
+parent
+- first-child {weight: 10}
+- second-child {weight: 20}
+- third-child {weight: 30}
+  - nested-child {weight: 10}
+```
+
+### Operators
+
+Set theory is applied when a new group is added as a child group. Three key concepts are at play: Intersections, Unions, and Differences.
+Any filters provided by the child groups are used to filter the members from the parent group using one of three operators: **Restrict (AND)**, **Include (OR)**, or **Exclude (NOT)**. This allows for logical parenthetical grouping of nested groups by the operator you choose for that child group association to the parent.
+
+We have attempted to simplify working with these operators by giving them both human-readable and Boolean name mappings. They are as follows:
+
+- **Restrict (Boolean `AND`)** - The **Restrict** operator performs a set _intersection_ on the queryset, and is equivalent to a Boolean `AND`. Any objects matching the child filter are _restricted_ (aka _intersected_) with the preceding filter. All filter criteria must match between the filters for a member object to be included in the resultant filter.
+- **Include (Boolean `OR`)** - The **Include** operator performs a set _union_ on the queryset, and is equivalent to a Boolean `OR`. Any objects matching the child filter are _included_ (aka _unioned_) with the preceding filter. Any filter criteria may match between the filters for member objects to be included in the resultant filter.
```

Comment:
Reading this, I feel the need to stir the can of worms a bit more - would `Extend` be a clearer verb than `Include`?

```suggestion
- **Include (Boolean `OR`)** - The **Include** operator performs a set _union_ on the queryset, and is equivalent to a Boolean `OR`. The preceding filter is _extended to include_ (aka _unioned with_) any objects matching the child filter. Any filter criteria may match between the filters for member objects to be included in the resultant filter.
```

Reply:
I don't know anymore. I've said the words so much they've lost all meaning!